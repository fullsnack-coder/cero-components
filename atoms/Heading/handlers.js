export const createHandlerBlur =
  ({ onBlur }) =>
  (event) => {
    onBlur(event)
  }
