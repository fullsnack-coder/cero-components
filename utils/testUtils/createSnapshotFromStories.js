import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
