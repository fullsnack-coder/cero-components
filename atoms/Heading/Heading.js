import React from 'react'
import PropTypes from 'prop-types'

import styles from './Heading.module.css'
import { options } from './constants'
import withStyles from '../../hocs/withStyles'

import { createHandlerBlur } from './handlers'

export const Heading = ({
  children,
  isCentered,
  isInline,
  isEditable,
  onBlur,
  getStyles,
}) => {
  return (
    <h1
      className={getStyles('heading', ['color', 'size', 'weight'], {
        'is-centered': isCentered,
        'is-inline': isInline,
      })}
      onBlur={createHandlerBlur({
        onBlur,
      })}
      contentEditable={isEditable}
    >
      {children}
    </h1>
  )
}

Heading.propTypes = {
  children: PropTypes.string.isRequired,
  getStyles: PropTypes.func.isRequired,
  isInline: PropTypes.bool,
  isCentered: PropTypes.bool,
  isEditable: PropTypes.bool,
  onBlur: PropTypes.func,
  color: PropTypes.oneOf(options.colors),
  size: PropTypes.oneOf(options.sizes),
  weight: PropTypes.oneOf(options.weights),
}

Heading.defaultProps = {
  color: 'base',
  size: 'md',
  weight: 'bold',
  isEditable: false,
  onBlur: () => {},
}

export default withStyles(styles)(Heading)
