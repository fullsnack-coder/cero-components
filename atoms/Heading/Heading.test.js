import * as stories from './Heading.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ atoms / Heading ]', () => {
  describe('#Heading', () => createSnapshotFromStories('Heading', stories))
})
