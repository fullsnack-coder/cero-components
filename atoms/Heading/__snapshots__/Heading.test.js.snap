// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ atoms / Heading ] #Heading when 'Heading' Centered is mounted should render 1`] = `
<DocumentFragment>
  <h1
    class="heading color-base size-md weight-bold is-centered"
  >
    Lose consequence night studies Wolves should wizards destruction burglar? All right, then. Keep your secrets.
  </h1>
</DocumentFragment>
`;

exports[`[ atoms / Heading ] #Heading when 'Heading' Colors is mounted should render 1`] = `
<DocumentFragment>
  <h1
    class="heading color-base size-md weight-bold"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
  <h1
    class="heading color-primary size-md weight-bold"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
  <h1
    class="heading color-tertiary size-md weight-bold"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
  <h1
    class="heading color-inverted size-md weight-bold"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
</DocumentFragment>
`;

exports[`[ atoms / Heading ] #Heading when 'Heading' Default is mounted should render 1`] = `
<DocumentFragment>
  <h1
    class="heading color-base size-md weight-bold"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
</DocumentFragment>
`;

exports[`[ atoms / Heading ] #Heading when 'Heading' Inline is mounted should render 1`] = `
<DocumentFragment>
  <h1
    class="heading color-base size-md weight-bold is-inline"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
</DocumentFragment>
`;

exports[`[ atoms / Heading ] #Heading when 'Heading' Sizes is mounted should render 1`] = `
<DocumentFragment>
  <h1
    class="heading color-base size-xs weight-bold"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
  <h1
    class="heading color-base size-sm weight-bold"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
  <h1
    class="heading color-base size-md weight-bold"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
  <h1
    class="heading color-base size-lg weight-bold"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
  <h1
    class="heading color-base size-xl weight-bold"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
  <h1
    class="heading color-base size-2xl weight-bold"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
</DocumentFragment>
`;

exports[`[ atoms / Heading ] #Heading when 'Heading' Weights is mounted should render 1`] = `
<DocumentFragment>
  <h1
    class="heading color-base size-md weight-normal"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
  <h1
    class="heading color-base size-md weight-bold"
  >
    Consent doubt Tuckborough challenge destroying.
  </h1>
</DocumentFragment>
`;
